tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,415 @@
+        result = ssh.command(
+            '''{{ echo '  :username: "{0}"';\
+            echo '  :password: "{1}"'; }} >> {2}'''.format(
+                settings.server.admin_username,
+                settings.server.admin_password,
+                HAMMER_CONFIG
+                )
+            )
+    return result.return_code
+
+
+@run_in_one_thread
+class HammerAuthTestCase(CLITestCase):
+    """Implements hammer authentication tests in CLI"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates users to be reused across tests"""
+        super(HammerAuthTestCase, cls).setUpClass()
+        cls.uname_admin = gen_string('alpha')
+        cls.uname_viewer = gen_string('alpha')
+        cls.password = gen_string('alpha')
+        cls.mail = 'test@example.com'
+        make_user({
+                'login': cls.uname_admin,
+                'password': cls.password,
+                'admin': '1',
+        })
+        make_user({
+                'login': cls.uname_viewer,
+                'password': cls.password,
+        })
+        User.add_role({'login': cls.uname_viewer, 'role': 'Viewer'})
+
+    @classmethod
+    def tearDownClass(cls):
+        """Making sure sessions are disabled after test run"""
+        configure_sessions(False)
+
+    @tier1
+    def test_positive_create_session(self):
+        """Check if user stays authenticated with session enabled
+
+        :id: fcee7f5f-1040-41a9-bf17-6d0c24a93e22
+
+        :Steps:
+
+            1. Set use_sessions, set short expiration time
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Wait until session expires, assert credentials
+               are required
+
+        :expectedresults: The session is successfully created and
+            expires after specified time
+        """
+        try:
+            idle = Settings.list({'search': 'name=idle_timeout'})[0][u'value']
+            Settings.set({'name': 'idle_timeout', 'value': 1})
+            result = configure_sessions()
+            self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+            Auth.login({
+                'username': self.uname_admin,
+                'password': self.password
+            })
+            result = Auth.with_user().status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            # list organizations without supplying credentials
+            with self.assertNotRaises(CLIReturnCodeError):
+                Org.with_user().list()
+            # wait until session expires
+            sleep(70)
+            with self.assertRaises(CLIReturnCodeError):
+                Org.with_user().list()
+            result = Auth.with_user().status()
+            self.assertIn(
+                LOGEDOFF_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+        finally:
```

Comment:
nitpick: i'd rather use `setting_cleanup()` to remove indentation.

Reply:
I'd rather keep it as it is for readability for the time being

Reply:
Looks like we have different vision of what's readable and what's not. For me overindented code looks less readable than a plain code with no extra indentation.

Reply:
@abalakh sure, I meant that if some future maintainer looks at the code, it will be immediately clear to him/her what the finally block is doing, without the need for checking the helper. I'd say this way of resetting settings prevails in current Robottelo. If we want to make it consistent, we can create a separate issue, but I think it shouldn't block this PR. What do you think?

Reply:
`setting_cleanup()` right after updating setting's value is pretty clear as for me too :) And regarding prevailing way - `setting_cleanup()` was introduced to replace try/finally, using outdated approach just because there're still some occurrences of it doesn't sound right for me (and i don't think that's the case, just checked and found 1 settings cleanup usage and 1 try/finally block).
Anyway, you're right, it's a nitpick and shouldn't block entire PR.